master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -89,14 +84,14 @@ public void updateRequestSlotsMeta(
     registeredShuffle.add(shuffleKey);
 
     String appId = Utils.splitShuffleKey(shuffleKey)._1;
-    appHeartbeatTime.compute(
+    applications.compute(
         appId,
-        (applicationId, oldTimestamp) -> {
-          long oldTime = System.currentTimeMillis();
-          if (oldTimestamp != null) {
-            oldTime = oldTimestamp;
-          }
-          return Math.max(System.currentTimeMillis(), oldTime);
+        (applicationId, applicationInfo) -> {
+          ApplicationInfo appInfo =
+              applicationInfo == null ? new ApplicationInfo() : applicationInfo;
+          appInfo.setHeartbeatTime(
+              Math.max(System.currentTimeMillis(), appInfo.getHeartbeatTime()));
```

Comment:
Here, and below, we can avoid an additional `System.currentTimeMillis`

```suggestion
          ApplicationInfo appInfo;
          
          if (null == applicationInfo) {
            appInfo = new ApplicationInfo();
          } else {
            appInfo = applicationInfo;
            appInfo.setHeartbeatTime(
              Math.max(System.currentTimeMillis(), appInfo.getHeartbeatTime()));
          }
```